### PR TITLE
build: use nox for tests and docs 

### LINKED ISSUE #239: use nox to build the documentation and the tests ?
I'm using this tool in my libs to build the documentation and tests in a isolated environment would you like me to implement it here ?

### PR DESCRIPTION
**Pull request by 12rambau**
Fix #239 

In this PR I added a few configurations to build unit tests and documentation pages using [nox](https://nox.thea.codes/en/stable/). It creates isolated environments to ensure that the build will work when send to RDT. It's also the first step toward unit testing. 

- create 2 nox sessions: "lint" to run the pre-commits (assuming it will be merged) and "docs" to create the documentation in a `docs/_build` folder.
- remove makefiles as they are nicely replaced by `nox -s docs`
- add the nox-produced files to `.gitignore`
- create a `doc` set of extra_requires to have everything needed for the documentation
- set up a .readthedocs.yaml file so that the documentation is built using the correct env

As I need to check the RDT build there will surely be test commits, I assume you will just squash my PR so I'm just sorry for the noise.

### CODE REVIEW

**Comment by 12rambau** [comment]
@maartenbreddels, @mariobuikhuizen I would like the doc to be build on main (at least latest) but when I install in dev mode there are many npm command that start. Of course it wasn't an issue before because RDT was build from the pipy version.
Would you have any guidance ?

**Comment by maartenbreddels** [comment]
I never build on rtd, because there wasn’t enough memory. Why do you want to do that?

**Comment by 12rambau** [comment]
If I want to work on the API build in #238 I would like to make sure latest is on track with main. So I would need to run: `pip install .[doc]`

**Comment by maartenbreddels** [comment]
I think you should install Jupyter lab to be able to build

**Comment by 12rambau** [comment]
thanks @maartenbreddels for the tips. The complete solution was "install jupyterlab before the lib to unable the build of jupyter extentions when `npm install` will be called from the js folder to execute webpack scripts".

with the pre-install job, it works like a charm. This is then ready for review

File: setup.py [hunk 1]
```diff
@@ -210,4 +210,13 @@ def run(self):
...
+    'extras_require': {
+        'doc': [
+            'sphinx',
+            'jupyter-sphinx',
+            'sphinx-rtd-theme',
+        ]
+    }
+}
+
+setup(**setup_args)
```
Comment (12rambau) [inline setup.py:222]:
```suggestion
    extras_require= {
        'doc': [
            'sphinx',
            'jupyter-sphinx',
            'sphinx-rtd-theme',
        ]
    }
)
```

**Comment by 12rambau** [comment]
sorry for the noise. I made many tests from RDT side. I hope they will find a solution from their side: https://github.com/readthedocs/readthedocs.org/issues/10223

**Comment by mariobuikhuizen** [comment]
This PR now depends on the fix in: https://github.com/widgetti/ipyvuetify/pull/259

**Comment by 12rambau** [comment]
@mariobuikhuizen I rebased on the current master and solved the remaining RDT related issue. I think this one is also ready to go.

File: docs/conf.py [hunk 2]
```diff
@@ -30,7 +30,8 @@
 # Add any Sphinx extension module names here, as strings. They can be
 # extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
 # ones.
-extensions = ["jupyter_sphinx.execute", "sphinx_rtd_theme"]
+
```
Comment (mariobuikhuizen) [inline docs/conf.py:33]: Why an extra new line?

**Review by mariobuikhuizen — CHANGES_REQUESTED** [review]
Looks good Thanks!

One small issue:

File: docs/conf.py (hunk 2 above)
Comment (12rambau) [reply docs/conf.py:33]: I did the merge in the gihub interface, so I didn't checked. Let me remove it

File: docs/conf.py [hunk 3]
```diff
@@ -30,7 +30,8 @@
...
+extensions = ["jupyter_sphinx", "sphinx_rtd_theme"]
```
Comment (12rambau) [inline docs/conf.py:34]:
```suggestion
extensions = ["jupyter_sphinx", "sphinx_rtd_theme"]
```

**Review by mariobuikhuizen — APPROVED** [review]

**Pull request merged**